pyvista/core/dataset.py

```diff
@@ -641,16 +641,27 @@ def set_active_scalars(self, name: Optional[str], preference='cell'):
             points or cells, it will prioritize an array matching this
             type.  Can be either ``'cell'`` or ``'point'``.
 
+        Returns
+        -------
+        pyvista.FieldAssociation
+            Association of the scalars matching ``name``.
+
+        numpy.ndarray
+            An array from the dataset matching ``name``.
+
         """
         if preference not in ['point', 'cell', FieldAssociation.CELL, FieldAssociation.POINT]:
             raise ValueError('``preference`` must be either "point" or "cell"')
         if name is None:
             self.GetCellData().SetActiveScalars(None)
             self.GetPointData().SetActiveScalars(None)
-            return
+            return FieldAssociation.NONE, np.array([])
         field = get_array_association(self, name, preference=preference)
         if field == FieldAssociation.NONE:
-            raise KeyError(f'Data named ({name}) is a field array which cannot be active.')
+            if name in self.field_data:
+                raise KeyError(f'Data named "{name}" is a field array which cannot be active.')
+            else:
+                raise KeyError(f'Data named "{name}" does not exist in this dataset.')
```

Comment:
Also not sure about these being `KeyError`s. We also have a `ValueError` about "could not be set as active scalars" a few lines below. So it's not even very consistent.

Reply:
Converted the first to `ValueError` in aee6ffa58ee0f26d183b464f761ef056f110cae2. It's not a key error because the key exists, it just has the wrong field association.